### ISSUE TITLE
MAJOR ISSUE: Epic not using standard system, instead using EPIC proprietary 

### ISSUE THREAD

**Issue by comorbidity** [smart-on-fhir/cumulus-library#96]
Epic doesn’t always the FHIR standard `system`, instead Epic uses custom `system=urn:oid:EPIC PROPRIATARY`.
This is why we now see that counts are “None” for Encounter type, serviceType, or priority. 
Probably also impacts Laboratory, Condition, and other FHIR resource types when system is expected to be standard.

SEE ALSO
https://github.com/smart-on-fhir/cumulus-library/issues/82
https://github.com/smart-on-fhir/cumulus-library/issues/31

IMPACT: 
**Coding.system** may not be used reliably in Epic EHR, which instead provides custom OID with no known public documentation. 
Consequently any programatic use that requires a standard "system" may not work in Epic sites. 

DEVIATION FROM STANDARD

**Encounter.type.system**
Expected 
http://hl7.org/fhir/us/core/STU4/ValueSet-us-core-encounter-type.html

Actual
`[{coding=[{code=2, display=Emergency, system=urn:oid:1.2.840.114350.1.13.93.2.7.10.698084.10110}], text=Emergency, id=null}, {coding=[{code=3, display=Hospital Encounter, system=urn:oid:1.2.840.114350.1.13.93.2.7.10.698084.30}], text=Hospital Encounter, id=null}, {coding=[{code=1, display=Emergency Admission, system=urn:oid:1.2.840.114350.1.13.93.2.7.10.698084.18875}], text=Emergency Admission, id=null}]
`

**Comment by comorbidity** [comment]
**Encounter.serviceType.system**

Expected 
http://hl7.org/fhir/R4/valueset-service-type.html

Actual 
urn:oid:1.2.840.114350.1.13.93.2.7.10.698084.18886

**Comment by comorbidity** [comment]
**Encounter.priority.system**

Expected 
_Field is optional, if field is provided system should be:_ 
https://terminology.hl7.org/1.0.0//ValueSet-v3-ActPriority.html

Actual 
`system=urn:oid:1.2.840.114350.1.13.93.2.7.10.698084.410`

**Comment by comorbidity** [comment]
**Observation Code (category=Laboratory)** 
Expected: LOINC required 
Actual: LOINC is usually provided but sometimes only Epic OID is provided instead 
`system=urn:oid:1.2.840.114350.1.13.93.2.7.2.768282`

**Comment by comorbidity** [comment]
**Condition.code** 
Expected: SNOMED-CT and/or ICD10 
Actual: frequent use of custom Epic OID for "reason for visit" effectively a text that looks like patient "chief complaint"
Impact: minimal, appears to only be related to free text fields. 
`system=urn:oid:1.2.840.114350.1.13.93.2.7.2.768282`

**Comment by comorbidity** [comment]
**Encounter.diagnosis**
Expected: `encounter-diagnosis` not `problem-list-item`
http://hl7.org/fhir/us/core/STU4/StructureDefinition-us-core-encounter.html

Actual: problem list was found to be linked from Encounter.diagnosis to a condition

**Comment by comorbidity** [comment]
POSSIBLE WORKAROUNDS: 
1. Do NOT filter patient cohorts using 
`select * from fhir_resource where system=.....`

2. Allow EPIC root OID http://oidref.com/1.2.840.114350 
`select * from fhir_resource where system like urn:oid:1.2.840.114350%`

**Comment by dogversioning** [comment]
confirmed the the root OID seems to be present across multiple implementations.

**Comment by dogversioning** [comment]
So, with the changes in #178, anyplace that we are looking at the value of system explictly, we are now looking for the observed EPIC OIDs. Between this and the [symptom extractor](https://github.com/smart-on-fhir/cumulus-system-extractor), we've got the spirit of this covered - we just know that, if we're looking for code systems, we will have to include some set of observed OIDs.

This will be a work in progress, but we've got the bones covered, so I'm going to close this issue as resolved.
